ModulemdTranslationHelpers/Utils.py

```diff
@@ -58,3 +63,94 @@ def get_translation_catalog_from_index(index, project_name):
         catalog.add(translatable_string, locations=locations)
 
     return catalog
+
+
+def get_modulemd_translations_from_catalog(catalogs, index):
+    # Nested dictionary `data` contains information from catalog like:
+    # Key_1: <module>:<stream>
+    # Value_1: Key_2: <locale>
+    #          Value_2: list of (<string_type>:<msg.string>)
```

Comment:
Instead of making a list here, we already know that the contents of this are going to match exactly to a `TranslationEntry`, so let's just create one of those instead. We won't be able to use a `defaultdict` though, because we need to construct the `TranslationEntry` with the correct locale (since this is not changeable after creation).